recipes/hklpy/meta.yaml

```diff
@@ -0,0 +1,51 @@
+{% set name = "hklpy" %}
+{% set version = "0.3.16" %}
+{% set sha256 = "bf3d0488bbe7f71d1d75ae791b871715dd311c3c3791510e5cc0a8cb063c6777" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  skip: true  # [py<36 or win or osx]
```

Comment:
This looks like it's pure python and could be `noarch: python`. If the only reason it's not is because the build fails due to `hkl` not being available on Windows/Mac, we can ignore those failures (noarch only builds on Linux). Being noarch drastically reduces the number of builds and makes it easier for the package to appear on future Python versions.

Reply:
That would be a mistake as core functionality is provided by that requirement.  Pinning to linux-64 ensures that no user on Mac or Windows or linux-32 will assume it can used.

Reply:
But isn't that already gated by the fact that `hkl` itself doesn't exist on those platforms?

Reply:
This was a decision for https://github.com/bluesky/hklpy/issues/108.  Should we reverse that now?

Reply:
Long-term (maybe a couple years away), the plan is to make the dependency on the *hkl* library optional.

Reply:
@dopplershift, thanks for the review and the useful suggestions! I see the reasoning behind your suggestion of using the `noarch` package for hklpy. However, the `hklpy` is a user-facing library, and the approach we took in the https://github.com/nsls-ii-forge feedstocks was to use the architecture-dependent packages for such user-facing libs, so that the dependencies for all intended Python versions are checked on the test step. I had quite a few occurrences when the conversion of noarch -> arch packages was needed due to incompatibility of the dependencies, that was hidden by that single build on Linux for a single Python version. See, e.g. https://github.com/nsls-ii-forge/anyio-feedstock/blob/master/recipe/meta.yaml, that was an attempt to resolve the mess with `dataclasses` and `typing-extensions`.

Overall, it's a trade-off that we have to decide on: (i) whether to build an arch-dependent package for Python 3.6-3.9 on Linux a few times a year (we don't tag that project too frequently), or (ii) have users complain about incompatible dependencies or the resolution conflicts. I personally choose the former (i).

Reply:
Thanks for the additional context. I'm happy to leave it non-`noarch` then.

Reply:
Thanks a lot!